src/lib/zf/attr.c

```diff
@@ -300,7 +316,9 @@ int zf_attr_get_str(struct zf_attr* attr, const char* name, char** val)
   switch( f->type ) {
   case zf_attr_type_str: {
     char** p = get_field_str(attr, f);
-    if( *p != NULL ) {
+    if( strcmp(f->name, "log_file")==0 )
```

Comment:
Just spotted the use of `f->name` here where the other bits just use `name` directly. Does this mean the other places will now be case-sensitive (as I assume `f->name` is the lowercase name and `name` could reasonably be something like `LOG_LEVEL`)?

Reply:
No, it means that I made an inconsistent (yet valid) change. I'll keep consistency and change it to `name`.

Reply:
My point was more that I feel like all of these should be `f->name` instead of just `name`. My understanding is that a user could call `zf_attr_get_str(attr, "LOG_FILE", &val)` and `zf_attr_info_find(name)` will match the appropriate attribute info (as it uses `strcasecmp` internally, so will match the lowercase string `"log_file"`). If you were to check `name` here in a case-sensitive way (`strcmp`) then something like `zf_attr_get_str(attr, "INTERFACE", &val)` would be valid, but `zf_attr_get_str(attr, "LOG_FILE", &val)` would not be.

Reply:
yes, I agree with you. I have overlooked `strcasecmp`. So all of these should be `f->name` to enforce matching the intended/appropriate attribute.